### PR TITLE
Move PanicHandlerPlugin into bevy_app

### LINKED ISSUE #12603: Move `PanicHandlerPlugin` into `bevy_app`
## What problem does this solve or what need does it fill?

A new crate was created in https://github.com/bevyengine/bevy/pull/12557: `bevy_panic_handler`. I spoke with @cart and @mockersf about this, and we agree that it makes more sense in `bevy_app` because:

1. Less administrative overhead to release the crate.
2. Less cognitive overhead for users and maintainers.

If and when this code becomes more significant, we can split it back out again.

## What solution would you like?

Move the code into a new module in `bevy_app`. Add the required optional dependencies. Get CI to pass.

## Additional context

This was discussed in #12557, but I didn't feel strongly enough to block forward progress: either solution was clearly better than `bevy_log`.

### PR DESCRIPTION
**Pull request by gabrielkryss**
# Objective

- Move `PanicHandlerPlugin` into `bevy_app`
- Fixes #12603 .

## Solution

- I moved the `bevy_panic_handler` into `bevy_app`
- Copy pasted `bevy_panic_handler`'s lib.rs into a separate module in `bevy_app` as a `panic_handler.rs` module file and added the `PanicHandlerPlugin` in lib.rs of `bevy_app`
- added the dependency into `cargo.toml` 

## Review notes

- I probably want some feedback if I imported App and Plugin correctly in `panic_handler.rs` line 10 and 11.
- As of yet I have not deleted `bevy_panic_handler` crate, wanted to get a check if I added it correctly.
- Once validated that my move was correct, I'll probably have to remove the panic handler find default plugins which I probably need some help to find.
- And then remove bevy panic_handler and making sure ci passes.
- This is my first issue for contributing to bevy so let me know if I am doing anything wrong.

## tools context
- rust is 1.76 version
- Windows 11

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
@theredfish can I get your review and feedback here? :)

File: crates/bevy_app/src/panic_handler.rs
Comment (alice-i-cecile) [inline crates/bevy_app/src/panic_handler.rs:1]: This is now a module, not a crate :)

File: crates/bevy_app/src/lib.rs
Comment (alice-i-cecile) [inline crates/bevy_app/src/lib.rs:27]: I don't think we need to export this in the prelude.

**Comment by alice-i-cecile** [comment]
Basic addition looks correct: feel free to remove the old version now. You should mark this PR as draft until you think it's ready to merge :)

I don't think we need to / want to remove this from `DefaultPlugins`: this is good default behavior for the web.

**Comment by alice-i-cecile** [comment]
Feel free to press "Resolve conversation" on comments that you've addressed as you complete their tasks. It helps reviewers (and you) track what work remains to be done.

**Comment by gabrielkryss** [comment]
image

I am not sure about the workspaces feature but I cannot find any more references to bevy_panic_handler, even in the root Cargo.toml. Maybe some advise on what I might have done wrong?

**Comment by gabrielkryss** [comment]
@theredfish seems like a manifest somewhere is expecting a `bevy_panic_handler` crate but I can't really find that reference even with grep

image

**Comment by theredfish** [comment]
> @theredfish seems like a manifest somewhere is expecting a `bevy_panic_handler` crate but I can't really find that reference even with grep

Did you try to `cargo clean` ? Locally I don't see your issue. However I can see the same issues as in the CI.

For CI seems connected to this [commit](https://github.com/bevyengine/bevy/commit/8327ce85d8c4ff9a7263f13b28065607fe524519) from the `bevy_ecs` crate. I get the same issues in the `main` branch by running 

```
cargo run -p ci -- compile
```

**Comment by gabrielkryss** [comment]
it seems its a cargo.toml thing in the parent dir maybe? but I don't see anything out of the ordinary inside the cargo.toml at the root dir of the project. 

After I deleted the `bevy_panic_handler` folder I also removed the dependencies in`bevy_internal`, referencing the above commit chain.

According to this: https://stackoverflow.com/questions/77848478/cargo-failing-to-load-manifest-for-workspace-member

**Comment by gabrielkryss** [comment]
> > @theredfish seems like a manifest somewhere is expecting a `bevy_panic_handler` crate but I can't really find that reference even with grep
> 
> Did you try to `cargo clean` ? Locally I don't see your issue. However I can see the same issues as in the CI.
> 
> For CI seems connected to this [commit](https://github.com/bevyengine/bevy/commit/8327ce85d8c4ff9a7263f13b28065607fe524519) from the `bevy_ecs` crate. I get the same issues in the `main` branch by running
> 
> ```
> cargo run -p ci -- compile
> ```

ya `cargo clean` yielded me the same result

|
image

**Comment by theredfish** [comment]
> ya `cargo clean` yielded me the same result

Do you have any entry with `bevy_panic_handler` in your lock file? 
The lock file is ignored so you will not break the branch by itself. What you can try is just to rename it with something like .backup at the end and let a new one to be generated.

> But anyway good to know that the CI is caused by the commit you mentioned.

Not necessarily caused by, but seems connected.

> And at least you can compile
Yes `cargo check` in `bevy_app` runs without any issue 👍

**Comment by bugsweeper** [comment]
> > > @theredfish seems like a manifest somewhere is expecting a `bevy_panic_handler` crate but I can't really find that reference even with grep
> > 
> > 
> > Did you try to `cargo clean` ? Locally I don't see your issue. However I can see the same issues as in the CI.
> > For CI seems connected to this [commit](https://github.com/bevyengine/bevy/commit/8327ce85d8c4ff9a7263f13b28065607fe524519) from the `bevy_ecs` crate. I get the same issues in the `main` branch by running
> > ```
> > cargo run -p ci -- compile
> > ```
> 
> ya `cargo clean` yielded me the same result and I see that you got the same CI error but not the failed to load manifest error. Hmm curious what happens if I just delete the cargo.lock but I am kinda scared to do so. but maybe I will try it. But anyway good to know that the CI is caused by the commit you mentioned. And at least you can compile the engine from the changes I made. At least that's what I understand.
> 
> image

Looks like you have problem with path, may be you should rename `bevy-fork` from you path to `bevy_fork` and try again?

**Comment by alice-i-cecile** [comment]
The diff looks good so far: can you update your PR description with the steps that need to be done still?

**Comment by theredfish** [comment]
On my side all good for the changes - I didn't send my approval since the PR is still in draft and CI failing. But once everything is fixed on the CI part, since it doesn't seem to be related to the code, I think it's ready to be merged.

**Comment by alice-i-cecile** [comment]
This [CI failure](https://github.com/bevyengine/bevy/actions/runs/8391120156/job/22980721162?pr=12640#step:6:2045) is real, but the Dependencies / check-advisories failure is spurious :) Once that's fixed, I'm also happy to approve and merge this!

**Comment by bugsweeper** [comment]
> > ya `cargo clean` yielded me the same result
> 
> Do you have any entry with `bevy_panic_handler` in your lock file? The lock file is ignored so you will not break the branch by itself. What you can try is just to rename it with something like .backup at the end and let a new one to be generated.
> 
> > But anyway good to know that the CI is caused by the commit you mentioned.
> 
> Not necessarily caused by, but seems connected. Also another PR experiences the same CI issues.
> 
> > And at least you can compile
> 
> Yes `cargo check` in `bevy_app` runs without any issue 👍

At this moment CI fails on doctests:
```
error: unresolved link to `crate::panic_handler::PanicHandlerPlugin`
 --> crates/bevy_internal/src/default_plugins.rs:4:30
  |
4 | /// * [`PanicHandlerPlugin`](crate::panic_handler::PanicHandlerPlugin)
  |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no item named `panic_handler` in module `bevy_internal`
  |
  = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(rustdoc::broken_intra_doc_links)]`
```
I think full `cargo run -p ci` should find all kinds of problem

**Comment by gabrielkryss** [comment]
> > > ya `cargo clean` yielded me the same result
> > 
> > 
> > Do you have any entry with `bevy_panic_handler` in your lock file? The lock file is ignored so you will not break the branch by itself. What you can try is just to rename it with something like .backup at the end and let a new one to be generated.
> > > But anyway good to know that the CI is caused by the commit you mentioned.
> > 
> > 
> > Not necessarily caused by, but seems connected. Also another PR experiences the same CI issues.
> > > And at least you can compile
> > 
> > 
> > Yes `cargo check` in `bevy_app` runs without any issue 👍
> 
> At this moment CI fails on doctests:
> 
> ```
> error: unresolved link to `crate::panic_handler::PanicHandlerPlugin`
>  --> crates/bevy_internal/src/default_plugins.rs:4:30
>   |
> 4 | /// * [`PanicHandlerPlugin`](crate::panic_handler::PanicHandlerPlugin)
>   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no item named `panic_handler` in module `bevy_internal`
>   |
>   = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
>   = help: to override `-D warnings` add `#[allow(rustdoc::broken_intra_doc_links)]`
> ```
> 
> @gabrielkryss Paths should be correct not only in code, but in document links also I think full `cargo run -p ci` should find all kinds of problem

Just got home, will take a look and correct that doc path reference once I settle in.

> The diff looks good so far: can you update your PR description with the steps that need to be done still?
> 
> I would do a project-wide search for `bevy_panic_handler` to find where the outstanding manifest entry is. Failing that, I would go over the diff of the PR that initially added this :)

I did a project-wide grep search and didn't see any more bevy_panic_handler symbols at least. Very curious about what is checking for this.

**Comment by gabrielkryss** [comment]
Good news! it seems like when I got back and ran `cargo run -p ci -- compile`, and it worked! No idea why and how but other than I just restarted my computer from last night I guess? 

Anyway, I made the doc correction and I found that when I run `cargo run -p ci -- doc` it times out at some point where it takes longer than 60 seconds for `bevy_gizmo` for me, maybe cuz I don't have a graphics card, and only have intel integrated graphics. So counting on the CI here.

**Comment by bugsweeper** [comment]
> Anyway, I made the doc correction and I found that when I run `cargo run -p ci -- doc` it times out at some point where it takes longer than 60 seconds for `bevy_gizmo` for me, maybe cuz I don't have a graphics card, and only have intel integrated graphics. So counting on the CI here.

Don't worry about that, this is just slow test (in my case also at least), if you wait it will finish.

**Comment by theredfish** [comment]
There is a PR in progress for the check advisories https://github.com/bevyengine/bevy/pull/12665 for info. Once it's merged we will be able to run the last failing job.

**Review by alice-i-cecile — APPROVED** [review]
I've run into the same issue with bevy_gizmo tests before: I should have an issue open for it currently :)

Anyways this looks good now! If you can resolve merge conflicts I'll merge this in for you.

**Review by theredfish — APPROVED** [review]

**Comment by gabrielkryss** [comment]
Thanks, everyone! 🚢 🚢 

This is my first open-source and bevy issue so I appreciate the help and hope to knock more of these out in the near future.

**Review by james7132 — APPROVED** [review]
It'd be nice to make this optional for those who want to minimize build sizes on wasm builds, but we can handle that in a follow-up PR.

**Comment by bugsweeper** [comment]
> If you can resolve merge conflicts I'll merge this in for you.

CI error looks weird
`cargo deny check advisories
  shell: /usr/bin/bash -e {0}
  env:
    CARGO_TERM_COLOR: always
    CARGO_INCREMENTAL: 0
2024-0[3](https://github.com/bevyengine/bevy/actions/runs/8398608533/job/23007232509?pr=12640#step:5:3)-23 07:16:[5](https://github.com/bevyengine/bevy/actions/runs/8398608533/job/23007232509?pr=12640#step:5:5)5 [ERROR] failed to fetch advisory database https://github.com/RustSec/advisory-db: An IO error occurred when talking to the server: error sending request for url (https://github.com/RustSec/advisory-db/info/refs?service=git-upload-pack)
Error: Process completed with exit code 1.`
Souldn't we just restart CI check?

**Comment by alice-i-cecile** [comment]
Can you resolve merge conflicts for us? Let us know if you need help; git can be a pain and we're happy to teach you.

**Comment by gabrielkryss** [comment]
> Can you resolve merge conflicts for us? Let us know if you need help; git can be a pain and we're happy to teach you.

Oh, my apologies I thought the merge conflicts were the responsibilities of someone else, my bad. I did a git pull on my branch but I don't get the conflict errors and it says only those with write access can merge pull requests I assume I don't have them hence I thought it was a person with a higher role that does them. 

So to clarify some instructions, do I do a rebase merge or merge commit to by main bevy fork branch and the PR will update?

Yeah maybe some clear instructions on how you guys deal with merge conflicts usually would be helpful as sometimes I just resolve them via the UI on GitHub which I don't think I can do in this case. Not really the best practice I know but its been working for me so far till now I guess. 😅

**Comment by alice-i-cecile** [comment]
Yep, totally understandable. I try and handle them when it's easy, but in more complex cases it's generally more efficient to ask  the author.

Personally, I use GitKraken for merge conflicts: I'll open Bevy up in it, and then merge the latest commit on `origin/main` into my branch. Once all the conflicts are resolved, I'll open it up in my IDE and make sure everything compiles and is formed. Then, I'll make the merge commit and push to my branch.

**Comment by theredfish** [comment]
From experience it's easier to do a merge commit in your own branch when you're not used to git. Then you should get conflicts to solve locally. Just be sure you updated your local main branch before.

- `git checkout main`
- `git pull --rebase` (just to avoid pull merge commit) 
- `git checkout your_branch`
- `git merge main`

Then you solve the conflicts if any and add the files then push. A merge commit will appear in your history and that's ok. Maintainers will merge and squash the commits into one (probably, depends of projects).

But if you're used to git and you are the only person working on the history you can rebase on main and then push force. But since it rewrites the history it can easily lead to issues. For someone not used to git I wouldn't recommend ☺️

**Comment by gabrielkryss** [comment]
thanks for the help guys I will get to it when I get back home and let you know if need more help, I try to use the CLI more recently and have been trying Lazygit into my workflow.

> From experience it's easier to do a merge commit in your own branch when you're not used to git. Then you should get conflicts to solve locally. Just be sure you updated your local main branch before.
> 
> * `git checkout main`
> * `git fetch upstream main` (given the source bevy repo is referenced as upstream. You can check with `git remote -v`)
> * `git pull --rebase upstream main` (`--rebase` is to avoid pull merge commit)
> 
> At this point `git log` will help you to check that the latest commit is the same as the one in the Bevy repo. You should see the graph information similar to this: `(HEAD -> main, upstream/main)` for the latest commit. It means your HEAD (local) and upstream reference the same commit.
> 
> * `git checkout your_branch`
> * `git merge main`
> 
> Then you solve the conflicts if any and add the files then push. A merge commit will appear in your history and that's ok. Maintainers will merge and squash the commits into one (probably, depends of projects).
> 

This is useful thanks for the instruction.

> But if you're used to git and you are the only person working on the history you can rebase on main and then push force. But since it rewrites the history it can easily lead to issues. For someone not used to git I wouldn't recommend ☺️

when you say only person working on the history you mean for a repo or for a PR? Since I think I am the only one who committed code here I kinda wanna try the rebase but most probably not gonna do that and just gonna follow your instructions but curious what the cli commands and process are for the rebase case.

**Comment by bugsweeper** [comment]
> when you say only person working on the history you mean for a repo or for a PR?

That meant for a repo, because if you force push while you miss some commits from server (if someone pushed work while you do your local changes), that commits can be lost. That mean you can reset someone work. If you are alone you can't erase anyone's else work.

**Comment by gabrielkryss** [comment]
> * `git checkout main`
> * `git fetch upstream main` (given the source bevy repo is referenced as upstream. You can check with `git remote -v`)

I've did these commands and got these output. From my understanding, I see that my main branch is referencing my `bevy-fork` repo. 

image

To make sure I am doing it right, to reference the bevy source dir I need to do `git remote add upstream <bevy_repo_url>` to add a remote called upstream on my main branch that references the bevy repo. 

Then do a `git fetch upstream` to fetch changes from source bevy repo.

And set the upstream to my main branch with `git branch -u upstream/main main`. If I am not wrong this makes a branch called `upstream/main` that references the remote `upstream` that references the source bevy repo. I might be wrong about this cuz idk if my 'main' branch still references my source bevy repo or it will or my `main` branch would still reference my `bevy-fork` repo and another branch called `upstream/main` will reference the source bevy repo.

And to verify I use `git branch -vv` that my branch probably `upstream/main` that it references the bevy repo

> * `git pull --rebase upstream main` (`--rebase` is to avoid pull merge commit)

and once I verify that my lastest commit in my local is the same as the latest in bevy engine repo. I can do the merge.

If I am right about that a branch called `upstream/main` is created that references the repo. then my merge line would look like below instead after I checkout to the branch with my changes of course.

```
git merge upstream/main
```

Hope this isn't too annoying I like to verbalize and verify what I do before I do it in case I make a mistake that is  annoying to save.

**Comment by gabrielkryss** [comment]
oh no I checked the commit and it doesn't seem like I did it quite right. The conflicts I resolved are minimal but the comit itself seems to be more than that. Is there any way I can resolve this?

**Comment by gabrielkryss** [comment]
So I did a bit of research and might have either or solution

1) Do a `git revert HEAD` and then push to remote
- This will make another commit to revert my previous commit and then I can redo the merge commit to resolve the conflicts properly.

2) Do a `git reset --hard HEAD~1` and then do a force push to remote
- This will reset my local changes to the previous commit and delete all the unstage changes, To not delete unstaged changes just do the command without the --hard flag.
- This will delete the previous commit, and then I can redo the merge commit to resolve the conflicts properly.

Let me know if this is valid or any other approach that I need to be aware of. And it seems like the CI is not even finishing and is stuck.

**Comment by bugsweeper** [comment]
> I don't see a lib.rs in my local version
> image

It could be a change in this Cargo.toml file since you started work, that's why there were a conflict for wich you returned  Cargo.toml inside last commit. Resolving conflicts - hard job, you should keep code from upstream if you didn't change it in this PR, and wisely combine upstreams and yours code if you changed it during making changes in this PR

**Comment by OneFourth** [comment]
You are correct, either `revert` or `reset` will work here, I would rather opt for the `reset` since it acts as a 'real' undo button, i.e. the state you were in before the commit is the state you are in after the `reset`, rather than `revert` which leaves both the original and the revert commits. One caveat is that it can get weird if the commit you're working on is a 'merge' commit, in which case it's easiest to do `git reset <hash> --hard` where hash is the last good commit.

Your commit isn't acting as a merge commit though, you can check this if you do
```
> git show --pretty=raw cb70fc500
commit cb70fc500b6d0ac194ec6fe885198afa815d6951
tree 7485e2645b3fb0f37b494df8101de3f6fb6a6868
parent 9881cf22daac469b21ab29147cc442e16cdbe899
...
```
You'll see that it only has one parent, rather than one of the other merge commits like this one that has two.
```
> git show --pretty=raw be4befaf6
commit be4befaf659a47c15dd301379133b920e9fbb180
tree f9c6ce7efb05f23f5f3e5ad35afcff934d4ddc9d
parent 74c18c0e2a9834263eb6507f0eccbfe6f1079c34
parent 7673afb03e8f0b3a7ff3cf5e1fc497b193050185
...
```
I'm not sure exactly what gets you into this state, but it was probably something like aborting the merge but keeping the changes and then committing them.

---

But anyway to give more explicit instructions (Sorry if you know some of this, just making sure), here's what I would do:
Remove the bad commit by setting ourselves to the previous commit
```
git reset HEAD~1 --hard
```
Fetch and then merge in the latest changes, in your case it would be
```
git fetch bevy_source_upstream
git merge bevy_source_upstream/main
```
Now you need to handle the merge conflicts, it looks like there would be two:
```
Auto-merging crates/bevy_app/Cargo.toml
Auto-merging crates/bevy_app/src/lib.rs
Auto-merging crates/bevy_app/src/panic_handler.rs
CONFLICT (content): Merge conflict in crates/bevy_app/src/panic_handler.rs
Auto-merging crates/bevy_internal/Cargo.toml
Auto-merging crates/bevy_internal/src/lib.rs
CONFLICT (modify/delete): crates/bevy_panic_handler/Cargo.toml deleted in HEAD and modified in bevy/main.  Version bevy/main of crates/bevy_panic_handler/Cargo.toml left in tree.
```
To fix them open up `crates/bevy_app/src/panic_handler.rs` and fix the conflict (In this case it looks like it's just adding in the bevy icons to the documentation (f096ad415 Set the logo and favicon for all of Bevy's published crates), so keep that and fix the "This crate provides..." comment, removing the merge conflict markers as well.
When you're done you can resolve the file with
```
git add crates/bevy_app/src/panic_handler.rs
```
Next we have `crates/bevy_panic_handler/Cargo.toml` which was deleted by us but modified by something else (72c51cdab Make `feature(doc_auto_cfg)` work), so we can just keep the deletion
```
git rm crates/bevy_panic_handler/Cargo.toml
```
Now you just have to commit the merge (`git commit` would also work here, but I like doing it this way so it tells me if I'm not 'in' a merge right now)
```
git merge --continue
```
And we're done! To verify we can check what files changed between main and your branch, this is before (Comparing main to what you have pushed currently) you can see a lot of files changed here which is a sign of a broken merge
```
> git diff --name-only bevy_source_upstream/main...origin/12603_move_panic_handler_plugin_into_bevy_app
.github/workflows/ci.yml
.github/workflows/docs.yml
Cargo.toml
assets/models/bunny.meshlet_mesh
crates/bevy_a11y/Cargo.toml
crates/bevy_a11y/src/lib.rs
crates/bevy_animation/Cargo.toml
crates/bevy_animation/src/animatable.rs
crates/bevy_animation/src/lib.rs
crates/bevy_app/Cargo.toml
...
```
Compared to our new fixed branch
```
> git diff --name-only bevy_source_upstream/main...HEAD
crates/bevy_app/Cargo.toml
crates/bevy_app/src/lib.rs
crates/bevy_app/src/panic_handler.rs
crates/bevy_internal/Cargo.toml
crates/bevy_internal/src/default_plugins.rs
crates/bevy_internal/src/lib.rs
crates/bevy_panic_handler/Cargo.toml
```

So finally, just do a `git push -f` (Always careful with this, we're the only ones working on this branch so it's fine here) to tell your origin to forget what it has and accept your new branch.
Hope this helps, let me know if anything is unclear and I'll explain further.

**Comment by RobWalt** [comment]
### Undo the changes

I don't have any working experience with `git revert` on merges (since I usually don't merge) so let's go with your second suggestion

> Do a `git reset --hard HEAD~1` and then do a force push to remote

In general I wouldn't recommend using the `--hard` flag as it can cause a lot of damage on accidents. So just go for `git reset HEAD~1`. That will keep all the changes in the unstaged area as you've already described it. In our situation we will only be left with the hundreds of changed files due to the merge and we want to get rid of it. If you're sure you want to undo all of these at once, then do:

- `git restore .` - undo all the stuff in the unstaged area
- `git clean -d -f .` - delete all new files that were created which are not tracked by git yet (be careful with this one as well in general!) 

That will undo your changes

### Update the main branch (optional)

I'd recommend a different work flow for updating your main branch. You already added the remote for the "original" bevy repo, which is basically step zero. To update the local main branch:

- `git switch main`
- `git pull upstream main` - this will pull the changes from the upstream main branch and will directly merge/rebase the changes into your local main branch (based on your git settings) 

and that's all. This way you don't need an extra branch for upstream main locally. I would only recommend that for very special cases where you want to work on a specific commit of the upstream repos branch.

### Rebasing

Let's just rebase with the freshly synced main:

- `git switch 12603_...`
- `git rebase main`

This will reapply all the changes from your branch onto main. Usually that means that you'll have to deal with more conflicts as each of your commits could have one. In general these conflicts are better managable though. In our case, git will say:

```git
interactive rebase in progress; onto 31d91466b
Last commands done (4 commands done):
   pick e911744e8 updated comment docs in panic_handler.rs
   pick 90243dae6 removed bevy_panic_handler_crate
  (see more in file .git/rebase-merge/done)
Next commands to do (3 remaining commands):
   pick ba85fe485 removed panic handler plugin from prelude export
   pick 74c18c0e2 removed bevy_panic_handler dependecy from bevy_internal
  (use "git rebase --edit-todo" to view and edit)
You are currently rebasing branch '12603_move_panic_handler_plugin_into_bevy_app' on '31d91466b'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add/rm <file>..." as appropriate to mark resolution)
        deleted by them: crates/bevy_panic_handler/Cargo.toml
        deleted by them: crates/bevy_panic_handler/src/lib.rs

no changes added to commit (use "git add" and/or "git commit -a")
```

you can just do the suggested `git restore --staged .`, since we want to keep the files. After that conflict is resolved, finish it up with:

- `git add .` (not really needed since after the restore we don't have any changes, but usually something like this is needed while rebasing)
- `git rebase --continue`

Normally if could happen that more conflicts appear. However, we're already done here and it will report a successful rebase. To finish this whole saga just do a final `git push --force-with-lease` (recommended over `git push --force` as it will do some safety checks, e.g. it checks if you would force push over another persons commits).

---

I hope that helped a bit, feel free to ask for more clarifications ✨

**Comment by gabrielkryss** [comment]
Thanks, @RobWalt and @OneFourth for the instructive write-up, really appreciate it! I think I will follow @OneFourth's instructions as I think I understand it more but I will try the force push with release. TBH, I'm kinda scared I'll mess it up again but hopefully, I won't haha 😅. Worst case maybe I will start another branch with another PR but hopefully, it won't come to that. Again, really appreciate the effort and time for the write-up.

**Comment by gabrielkryss** [comment]
I did the merge but it seems the ci (ci and ubuntu jobs) has these errors related to the icon stuff added to the panic handler module. It seems like the macro works for crates but not modules. Should I remove it then?

image

**Comment by OneFourth** [comment]
That makes sense to remove it then since it's not a crate anymore, the conflicts look good otherwise!

**Comment by gabrielkryss** [comment]
thanks for the help everyone, looks like it's good to go, CI seems to be all green.

**Comment by alice-i-cecile** [comment]
Awesome, merging. Thanks for making this, and thank you all so much for helping out our new contributor :)

**Pull request merged**
